### PR TITLE
Fixed background logging issue by replacing zmq handler with a StreamHandler

### LINKED ISSUE #1305: Background Helper: Logging in background does not work

### PR DESCRIPTION
**Pull request by marc-tonsen**
Fixes #1305

### CODE REVIEW

File: pupil_src/shared_modules/background_helper.py [hunk 1]
```diff
@@ -56,6 +58,14 @@ def _wrapper(self, pipe, _should_terminate_flag, generator, *args, **kwargs):
             pipe.close()
             logger.debug('Exiting _wrapper')
 
+    def _fix_background_logger(self):
```
[the remark this answers is not in the document]
Comment (marc-tonsen) [reply pupil_src/shared_modules/background_helper.py]: Good idea. @papr can you write add a comment in the code or here? You understand much better what is going on here.

File: pupil_src/shared_modules/background_helper.py [hunk 2]
```diff
@@ -56,6 +58,14 @@ def _wrapper(self, pipe, _should_terminate_flag, generator, *args, **kwargs):
...
+        del logger.root.handlers[0]
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter(
```
Comment (mkassner) [inline pupil_src/shared_modules/background_helper.py]: so this streams log messages to STDOUT, would it not be better to send the log messages through  the zmq IPC just like all other parts of pupil do? Or am i missing something? 

Alternative would be to send all log messages through the wrapper pipe... and then log them in the mainthread (which will also put them on the IPC.)

Comment (marc-tonsen) [reply pupil_src/shared_modules/background_helper.py]: I think you are right, @papr suggested this as well yesterday.

Comment (papr) [reply pupil_src/shared_modules/background_helper.py]: Yes, there is a nice solution which I will push into the PR later today.

**Comment by papr** [comment]
I pushed a series of commits. Relevant commit for review is cc681842615dbc170415afabf8ea909d2c6d0d33

**Comment by papr** [comment]
@marc-tonsen @ckbaumann Please review when you have time.

Please adopt your PRs #1268 and #1322 s.t. they use the new `IPC_Logging_Task_Proxy`.

**Comment by marc-tonsen** [comment]
When switching from `Task_Proxy` to `IPC_Logging_Task_Proxy` I get the following error in #1268 .

`Process <function video_processing_generator at 0x7f95aff471e0>:
Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/cluster/users/Marc/code/pupil-software/pupil/pupil_src/shared_modules/background_helper.py", line 123, in _wrapper
    self._enforce_IPC_logging(ipc_push_url)
  File "/cluster/users/Marc/code/pupil-software/pupil/pupil_src/shared_modules/background_helper.py", line 135, in _enforce_IPC_logging
    handler = zmq_tools.ZMQ_handler(zmq_ctx, ipc_push_url)
  File "/cluster/users/Marc/code/pupil-software/pupil/pupil_src/shared_modules/zmq_tools.py", line 37, in __init__
    self.socket = Msg_Dispatcher(ctx, ipc_push_url)
  File "/cluster/users/Marc/code/pupil-software/pupil/pupil_src/shared_modules/zmq_tools.py", line 171, in __init__
    self.socket.connect(url)
  File "zmq/backend/cython/socket.pyx", line 580, in zmq.backend.cython.socket.Socket.connect
  File "zmq/backend/cython/checkrc.pxd", line 25, in zmq.backend.cython.checkrc._check_rc
zmq.error.ZMQError: Invalid argument`

**Comment by papr** [comment]
Did you pass the extra argument? See the changed files of this pr for an example.

**Pull request merged**